### PR TITLE
Code improvements including: removed unnecessary modules, improved type definitions, improved module naming

### PR DESCRIPTION
**Pull request by raegen**
- cleanup unused/unneeded modules, add more types, export typedoc documentation for wiki, rename modules to improve verbosity: 
  - server -> api, 
  - inbox -> dispatcher,
  - debug-> logger,
  - auto_track -> tracking,
  - microdata -> metadata,
* add typedoc module definitions, make event metadata consistent with documentation

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+ */
+
+import logger from "./logger";
+import { version } from "../package.json";
+import { Event } from "./event";
+
+const toDigits = (d: number, n: number): string =>
+  ("0" + Math.abs(n)).slice(-d);
+
+export class DateTime extends Date {
```
Comment (nordfjord) [inline src/api.ts]: I'm not quite sure why this is needed.

`Date` has `.toISOString()` and `.toJSON()` methods already

**Comment by nordfjord** [comment]
I actually think Microdata is more descriptive than Metadata, https://en.wikipedia.org/wiki/Microdata_(HTML)

File: src/partner.ts [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
 import { UTM } from "./campaign";
 
 export const PARTNER_ID_KEY = "__as.partner_id";
 export const PARTNER_SID_KEY = "__as.partner_sid";
 
 const updatePartnerInfo = () => {
-  const uri = parser.parseURI(window.location.href);
-  let partnerId = uri.queryKey[PARTNER_ID_KEY];
-  let partnerSId = uri.queryKey[PARTNER_SID_KEY];
+  const uri = new URL(window.location.href);
```
[the remark this answers is not in the document]
Comment (raegen) [reply src/partner.ts]: You're correct, it literally didn't even cross my mind.

**Comment by raegen** [comment]
@nordfjord about that toJSON, it's removed, I've moved it to a util fn, but it was there because we have a custom string format. It looks like ISO but it's not, it's not UTC but local and has the offset at the end.

**Comment by nordfjord** [comment]
ISO 8601 does allow offsets

`2018-05-30T16:37:28+00:00` is legal `2018-05-30T16:37:28+01` is legal and `2018-05-30T16:37:28+0100` is legal too

**Comment by nordfjord** [comment]
Oh, but alas, toISOString always coerces to UTC time :(, I'll get off my high horse

**Comment by raegen** [comment]
@nordfjord haha, no need, you didn't state anything incorrect :D The toISOString implementation is questionable. Cheers

**Pull request merged**